dotnet/src/Skills/Skills.OpenAPI/Extensions/KernelAIPluginExtensions.cs

```diff
@@ -0,0 +1,411 @@
+using System.Text.Json.Nodes;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+using Microsoft.SemanticKernel.Skills.OpenAPI.Model;
+using Microsoft.SemanticKernel.Skills.OpenAPI.OpenApi;
+
+namespace Microsoft.SemanticKernel.Skills.OpenAPI.Extensions;
+
+public static class KernelAIPluginExtensions
+{
+    /// <summary>
+    /// Imports an AI plugin that is exposed as an OpenAPI v3 endpoint or through OpenAI's ChatGPT format.
+    /// </summary>
+    /// <param name="kernel">Semantic Kernel instance.</param>
+    /// <param name="skillName">Skill name.</param>
+    /// <param name="filePath">The file path to the AI Plugin</param>
+    /// <param name="executionParameters">Skill execution parameters.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns></returns>
+    public static async Task<IDictionary<string, ISKFunction>> ImportAIPluginAsync(
+        this IKernel kernel,
+        string skillName,
+        string filePath,
+        OpenApiSkillExecutionParameters? executionParameters = null,
+        CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(kernel);
+        Verify.ValidSkillName(skillName);
+
+#pragma warning disable CA2000 // Dispose objects before losing scope. No need to dispose the Http client here. It can either be an internal client using NonDisposableHttpClientHandler or an external client managed by the calling code, which should handle its disposal.
+        var internalHttpClient = HttpClientProvider.GetHttpClient(kernel.Config, executionParameters?.HttpClient, kernel.Log);
+#pragma warning restore CA2000 // Dispose objects before losing scope. No need to dispose the Http client here. It can either be an internal client using NonDisposableHttpClientHandler or an external client managed by the calling code, which should handle its disposal.
+
+        var pluginJson = await LoadJsonFromFilePath(
+            kernel,
+            filePath,
+            executionParameters,
+            internalHttpClient,
+            cancellationToken).ConfigureAwait(false);
+
+        if (TryParseAIPluginForUrl(pluginJson, out var openApiUrl))
+        {
+            return await kernel
+                .ImportAIPluginAsync(
+                    skillName,
+                    new Uri(openApiUrl),
+                    executionParameters,
+                    cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
+        }
+
+        return await LoadSkill(
+            kernel,
+            skillName,
+            executionParameters,
+            internalHttpClient,
+            pluginJson,
+            cancellationToken).ConfigureAwait(false);
```

Comment:
The code starting on line 56 and ending at line 73 is repeated three times in this file. Consider moving it to a separate private method to avoid duplication.

Reply:
Cleaned up as suggested